From 8f0e84bec513ad28c3ec479053fbc59add3959e1 Mon Sep 17 00:00:00 2001
From: Pino Toscano <toscano.pino@tiscali.it>
Date: Sat, 27 Dec 2014 17:32:51 +0100
Subject: [PATCH] small improvements to the hurd-g++ mkspec

- remove the X11R6 paths, since they were gone for years, and their
  lack may cause build issues with CMake config files
- add empty QMAKE_*_XCB variables, as done in the common linux.conf
- add to QMAKE_PLATFORM, not just reset it (just done in other mkspecs)
- borrow QMAKE_LFLAGS_GCSECTIONS from the common linux.conf

Change-Id: I94e05032f8195bbda73dffe1da02eec7ac679045
Reviewed-by: Thiago Macieira <thiago.macieira@intel.com>
---
 mkspecs/hurd-g++/qmake.conf | 15 ++++++++++-----
 1 file changed, 10 insertions(+), 5 deletions(-)

diff --git a/mkspecs/hurd-g++/qmake.conf b/mkspecs/hurd-g++/qmake.conf
index 54a152d..d53b49c 100644
--- a/mkspecs/hurd-g++/qmake.conf
+++ b/mkspecs/hurd-g++/qmake.conf
@@ -3,19 +3,20 @@
 #
 
 MAKEFILE_GENERATOR      = UNIX
-QMAKE_PLATFORM          = hurd
+QMAKE_PLATFORM         += hurd
 CONFIG                 += incremental
 QMAKE_INCREMENTAL_STYLE = sublib
 
 QMAKE_CFLAGS_THREAD    += -D_REENTRANT
 QMAKE_CXXFLAGS_THREAD  += $$QMAKE_CFLAGS_THREAD
+QMAKE_LFLAGS_GCSECTIONS = -Wl,--gc-sections
 
 QMAKE_INCDIR            =
 QMAKE_LIBDIR            =
-QMAKE_INCDIR_X11        = /usr/X11R6/include
-QMAKE_LIBDIR_X11        = /usr/X11R6/lib
-QMAKE_INCDIR_OPENGL     = /usr/X11R6/include
-QMAKE_LIBDIR_OPENGL     = /usr/X11R6/lib
+QMAKE_INCDIR_X11        =
+QMAKE_LIBDIR_X11        =
+QMAKE_INCDIR_OPENGL     =
+QMAKE_LIBDIR_OPENGL     =
 QMAKE_INCDIR_OPENGL_ES2 = $$QMAKE_INCDIR_OPENGL
 QMAKE_LIBDIR_OPENGL_ES2 = $$QMAKE_LIBDIR_OPENGL
 QMAKE_INCDIR_EGL        =
@@ -33,6 +34,10 @@ QMAKE_LIBS_OPENGL_ES2   = -lGLESv2
 QMAKE_LIBS_OPENVG       = -lOpenVG
 QMAKE_LIBS_THREAD       = -lpthread
 
+QMAKE_CFLAGS_XCB        =
+QMAKE_LIBS_XCB          =
+QMAKE_DEFINES_XCB       =
+
 QMAKE_AR                = ar cqs
 QMAKE_OBJCOPY           = objcopy
 QMAKE_NM                = nm -P
-- 
2.1.4

